### PR TITLE
Fix failed MOI tests

### PR DESCRIPTION
**Pull request by metab0t**
I applied the same tricks used in Xpress.jl here to fix 3 failed MOI tests.
The `test_solve_conflict_feasible` fails because Gurobi cannot compute IIS on a feasible model

### CODE REVIEW

File: src/MOI_wrapper/MOI_wrapper.jl [hunk 1]
```diff
@@ -3409,7 +3417,7 @@ function MOI.get(model::Optimizer, ::MOI.ListOfConstraintTypesPresent)
 end
 
 function MOI.get(model::Optimizer, ::MOI.ObjectiveFunctionType)
-    if model.is_feasibility
+    if !model.is_objective_set
```
Comment (odow) [inline src/MOI_wrapper/MOI_wrapper.jl:3420]: I think in other wrappers I removed the `return nothing`. If the objective isn't set, getting `MOI.ObjectiveFunctionType` doesn't have a valid return, so can return anything.

**Comment by odow** [comment]
Can you fix the format-check please?

File: src/MOI_wrapper/MOI_wrapper.jl (hunk 1 above)
Comment (metab0t) [reply src/MOI_wrapper/MOI_wrapper.jl]: Done

File: src/MOI_wrapper/MOI_wrapper.jl [hunk 2]
```diff
@@ -661,7 +662,13 @@ function MOI.get(model::Optimizer, ::MOI.ListOfVariableAttributesSet)
...
-    attributes = Any[MOI.ObjectiveSense()]
+    if MOI.is_empty(model)
+        return Any[]
+    end
+    attributes = Any[]
+    if model.objective_sense !== nothing
+        push!(attributes, MOI.ObjectiveSense())
+    end
     typ = MOI.get(model, MOI.ObjectiveFunctionType())
     if typ !== nothing
```
Comment (odow) [inline src/MOI_wrapper/MOI_wrapper.jl:673]: Is this passing tests? It should be
```Julia
if model.is_objective_set
    F = MOI.get(model, MOI.ObjectiveFunctionType())
    push!(attributes, MOI.ObjectiveFunction{F}())
end
```
(ObjectiveFunctionType now never returns `nothing`.)

Comment (metab0t) [reply src/MOI_wrapper/MOI_wrapper.jl]: Yes, tests passed even if this is not corrected. Maybe a new MOI test should be added?

**Comment by metab0t** [comment]
I think `objective_type` should be a nullable type like `objective_sense`. 

`is_objective_set` can be discarded and `objective_type = nothing` represents no objective function is set.

The complicated thing is here. We need to check the corresponding variable of changed coefficient and set `objective_type` if `objective_type = _SINGLE_VARIABLE` currently.

```julia
function MOI.modify(
    model::Optimizer,
    ::MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}},
    chg::MOI.ScalarCoefficientChange{Float64},
)
    ret = GRBsetdblattrelement(
        model,
        "Obj",
        Cint(column(model, chg.variable) - 1),
        chg.new_coefficient,
    )
    _check_ret(model, ret)
    model.is_objective_set = true
    _require_update(model)
    return
end
```

**Comment by odow** [comment]
I think this is okay for now.

**Pull request merged**